django/thunderstore/repository/comment.py

```diff
@@ -0,0 +1,80 @@
+from django import forms
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+
+from thunderstore.community.models import PackageListing
+from thunderstore.repository.models import Comment
+
+
+def clean_content(content: str) -> str:
+    return content.strip()
+
+
+class CreateCommentForm(forms.ModelForm):
+    class Meta:
+        model = Comment
+        fields = ["content"]
+
+    def __init__(
+        self,
+        user: User,
+        commented_object: PackageListing,
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.user = user
+        self.commented_object = commented_object
+
+    def clean_content(self) -> str:
+        return clean_content(self.cleaned_data["content"])
+
+    def save(self, *args, **kwargs) -> Comment:
+        self.instance.author = self.user
+        self.instance.commented_object = self.commented_object
+        return super().save(*args, **kwargs)
+
+
+class EditCommentForm(forms.Form):
+    content = forms.CharField(max_length=Comment._meta.get_field("content").max_length)
+    is_pinned = forms.BooleanField(required=False)
+
+    def __init__(self, user: User, comment: Comment, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.user = user
+        self.comment = comment
+
+    def can_edit_content(self) -> bool:
+        # Only the comment author can edit a comment's content
+        return self.user == self.comment.author
+
+    def can_pin(self) -> bool:
+        commented_object = self.comment.commented_object
+        if isinstance(commented_object, PackageListing):
+            # Must be a member of the identity to pin
+            return commented_object.package.owner.members.filter(
+                user=self.user,
+            ).exists()
+        raise NotImplementedError()
```

Comment:
Maybe `False` would be a better default, that just makes it so nobody can pin comments on objects where we haven't implemented the feature yet?

Reply:
What do you think about it being a way to easily flag the issue in tests/testing? Forgetting to implement the type here and not writing a test for that specific parent type (expecting it to "just work" as it is already tested with other parent types) could cause the default to be left when being pushed to prod, but with an explicit error you can use it as a reminder to implement the type there and if you try it in testing you'll get an error which will be clear even if you aren't paying complete attention.

Reply:
During tests it would be nice to have fail indeed, but during runtime maybe not so much. Maybe we could have a helper that raises in test environments but logs to sentry in production environments, e.g:
```python
def capture_exception(exception):
    testing = bool("PYTEST_CURRENT_TEST" in os.environ)
    if testing or settings.NO_SILENT_FAIL:
        raise exception
    if not testing:
        capture_sentry_exception(exception)
```